templates/coredns/0/docker-compose.yml

```diff
@@ -0,0 +1,31 @@
+coredns:
+  image: coredns/coredns:1.0.1
+  labels:
+    io.rancher.sidekicks: data
+    io.rancher.container.hostname_override: container_name
+  command: [ "-conf", "/etc/coredns/Corefile" ]
+  ports:
+    - 53:53/tcp
+    - 53:53/udp
```

Comment:
What do you think if you add a question `PUBLISH_PORT`, with default value of 53, to be able to publish the service in configurable port??
```
-    - 53:53/tcp
-    - 53:53/udp
+    - ${PUBLISH_PORT}:${PUBLISH_PORT}/tcp
+    - ${PUBLISH_PORT}:${PUBLISH_PORT}/udp
```

Also a modification at line 1 of `https://github.com/Jason-ZW/Dockerfile/blob/master/coredns/Corefile.tmpl` would be needed.
```
-.:53 {
+.:{{ getenv "PUBLISH_PORT" }} {
```